pkg/operation/botanist/component/logging/fluentoperator/fluent_operator.go

```diff
@@ -0,0 +1,501 @@
+	for _, clusterOutput := range clusterOutputs {
+		operatorCustomResources = append(operatorCustomResources, clusterOutput)
+	}
+
+	for _, clusterInput := range f.additionalInputs {
+		operatorCustomResources = append(operatorCustomResources, clusterInput)
+	}
+
+	for _, clusterFilter := range f.additionalFilters {
+		operatorCustomResources = append(operatorCustomResources, clusterFilter)
+	}
+
+	for _, clusterParser := range f.additionalParsers {
+		operatorCustomResources = append(operatorCustomResources, clusterParser)
+	}
+
+	serializedOperatorResources, err := operatorRegistry.AddAllAndSerialize(operatorResources...)
+	if err != nil {
+		return err
+	}
+
+	serializedOperatorCustomResources, err := operatorCustomResourcesRegistry.AddAllAndSerialize(operatorCustomResources...)
+	if err != nil {
+		return err
+	}
+
+	if err := managedresources.CreateForSeed(ctx, f.client, f.namespace, CustomResourcesManagedResourceName, false, serializedOperatorCustomResources); err != nil {
+		return err
+	}
+
+	return managedresources.CreateForSeed(ctx, f.client, f.namespace, OperatorManagedResourceName, false, serializedOperatorResources)
+}
+
+func (f *fluentOperator) Destroy(ctx context.Context) error {
+	if err := managedresources.DeleteForSeed(ctx, f.client, f.namespace, CustomResourcesManagedResourceName); err != nil {
+		return err
+	}
+
+	return managedresources.DeleteForSeed(ctx, f.client, f.namespace, OperatorManagedResourceName)
+}
+
+// TimeoutWaitForManagedResources is the timeout used while waiting the managed resources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResources = 5 * time.Minute
+
+func (f *fluentOperator) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResources)
+	defer cancel()
+
+	if err := managedresources.WaitUntilHealthy(timeoutCtx, f.client, f.namespace, CustomResourcesManagedResourceName); err != nil {
+		return err
+	}
+
+	return managedresources.WaitUntilHealthy(timeoutCtx, f.client, f.namespace, OperatorManagedResourceName)
+}
+
+func (f *fluentOperator) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResources)
+	defer cancel()
+
+	if err := managedresources.WaitUntilDeleted(timeoutCtx, f.client, f.namespace, CustomResourcesManagedResourceName); err != nil {
+		return err
+	}
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, f.client, f.namespace, OperatorManagedResourceName)
+}
+
+func (f *fluentOperator) computeEnv() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name: "NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.namespace",
+				},
+			},
+		},
+	}
+}
+
+func (f *fluentOperator) computeVolumes() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			Name:      "env",
+			MountPath: "/fluent-operator",
+		},
+	}
+}
```

Comment:
Drop this and inline it above. There is no benefit in having dedicated functions.